### PR TITLE
[SPARK-41328][CONNECT][PYTHON][FOLLOW-UP] Simplify startsWith and endsWith

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

1. `startsWith` and `endsWith` can be implemented by simplify invoking `_bin_op` with additional documentation.
2.  Remove not working examples in the documentation.

### Why are the changes needed?

Codebase simplification. 

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

Existing UT.

### CODE REVIEW

**Comment by amaliujia** [comment]
cc @zhengruifeng

File: python/pyspark/sql/connect/column.py [hunk 1]
```diff
@@ -508,45 +508,24 @@ def contains(self, other: Union[PrimitiveType, "Column"]) -> "Column":
...
-
-        Parameters
-        ----------
-        other : :class:`Column` or str
-            string at start of line (do not use a regex `^`)
-
-        Examples
-        --------
-        >>> df = spark.createDataFrame(
-        ...      [(2, "Alice"), (5, "Bob")], ["age", "name"])
```
Comment (amaliujia) [inline python/pyspark/sql/connect/column.py:523]: This does not work in current Connect `createDataFrame` which does not allow a `schema`

Comment (zhengruifeng) [reply python/pyspark/sql/connect/column.py:523]: what about just removing the examples, if we can not support them now.

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/connect/column.py (hunk 1 above)
Comment (amaliujia) [reply python/pyspark/sql/connect/column.py:523]: A bit messy on our examples. Ideally every examples should be tested and proven to work....

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
merged into master

**Pull request closed without merging**